compiler/rustc_ast/src/util/literal.rs

```diff
@@ -196,9 +202,32 @@ impl LitKind {
 }
 
 impl MetaItemLit {
-    /// Converts token literal into a meta item literal.
+    /// Converts a token literal into a meta item literal.
     pub fn from_token_lit(token_lit: token::Lit, span: Span) -> Result<MetaItemLit, LitError> {
-        Ok(MetaItemLit { token_lit, kind: LitKind::from_token_lit(token_lit)?, span })
+        Ok(MetaItemLit {
+            symbol: token_lit.symbol,
+            suffix: token_lit.suffix,
+            kind: LitKind::from_token_lit(token_lit)?,
+            span,
+        })
+    }
+
+    /// Cheaply converts a meta item literal into a token literal.
+    pub fn as_token_lit(&self) -> token::Lit {
```

Comment:
Basically, the only non-suspicious uses of `synthesize_token_lit` and this function are AST/HIR pretty-printing (including rustfmt), and recovering a token from `ExprKind::IncludedBytes`.

Reply:
I ended up removing `synthesize_token_lit`.